### ISSUE TITLE
New version: Package subpath './vite' is not defined by "exports"

### ISSUE THREAD

**Issue by DutchEllie** [sveltejs/kit#5454]
### Describe the bug

As the name implies. I updated the svelte-kit package and now I have to use Vite. Now, I found 1 instructional video online on how to actually transition to the new version. After following it, my package building and dev fails. with the following error:

```
failed to load config from <repo path>/vite.config.js
error when starting dev server:
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './vite' is not defined by "exports" in <repo path>/node_modules/@sveltejs/kit/package.json imported from <repo path>/vite.config.js
    at new NodeError (node:internal/errors:388:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:440:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:719:3)
    at packageResolve (node:internal/modules/esm/resolve:893:14)
    at moduleResolve (node:internal/modules/esm/resolve:959:20)
    at defaultResolve (node:internal/modules/esm/resolve:1174:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:605:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:318:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)
    at link (node:internal/modules/esm/module_job:78:36)
```

My `vite.config.js` file looks like this:

```
import { sveltekit } from '@sveltejs/kit/vite';

/** @type {import('vite').UserConfig} */
const config = {
        plugins: [sveltekit()]
};

export default config;
```

Which is identical to the one in the CLI error before upgrading.

This is what my `package.json` looks like (shortened a little bit)

```
"scripts": {
	"dev": "vite dev",
	"build": "vite build",
	"preview": "vite preview"
}
```

What is going wrong here?

### Reproduction

Try to update an existing app by following the instructions.
1. Installing vite:
```
$ sudo npm install -g vite
```
2. Creating `vite.config.js` by creating the file in the project's root and copying the example file shown in the error message
3. Updating the `package.json` file with the new commands like above.
4. Run `npm run dev` or `npm run build` or `npm run preview`

### Logs

```shell
failed to load config from <repo path>/vite.config.js
error when starting dev server:
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './vite' is not defined by "exports" in <repo path>/node_modules/@sveltejs/kit/package.json imported from <repo path>/vite.config.js
    at new NodeError (node:internal/errors:388:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:440:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:719:3)
    at packageResolve (node:internal/modules/esm/resolve:893:14)
    at moduleResolve (node:internal/modules/esm/resolve:959:20)
    at defaultResolve (node:internal/modules/esm/resolve:1174:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:605:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:318:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)
    at link (node:internal/modules/esm/module_job:78:36)
```
```


### System Info

```shell
System:
    OS: Linux 5.18 Arch Linux
    CPU: (16) x64 AMD Ryzen 7 PRO 5850U with Radeon Graphics
    Memory: 5.11 GB / 14.47 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.4.0 - /usr/bin/node
    npm: 8.11.0 - /usr/bin/npm
  Browsers:
    Firefox: 102.0
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.53 
    @sveltejs/adapter-node: ^1.0.0-next.78 => 1.0.0-next.78 
    @sveltejs/kit: next => 1.0.0-next.357 
    svelte: ^3.44.0 => 3.48.0 
    vite: ^2.9.14 => 2.9.14
```

### Severity

blocking all usage of SvelteKit

### Additional Information

Without being rude, I think it's pretty weird that a package update cannot migrate itself. Is it normal (I am new to JS and stuff) to have to manually edit config files when a breaking update happens?

**Comment by Conduitry** [comment]
If you found a tutorial video telling you to install Vite globally with `sudo npm install -g vite`, that's absolutely bad advice. `vite` should be a devdependency in your project. You can take a look at what's now created with a new `npm create svelte` to get an idea for what you need to change in your project.

**Comment by DutchEllie** [comment]
I only used `sudo npm install -g vite` because I could not execute the `vite` command when I did just do `npm install --save-dev vite` which is what the video told me to do. I did that as well, which resulted in an issue with `vite` not being found, which is why I did the global install. I also used `sudo` because the global install failed with a permission issue.

I have taken a look at the template like you said and I do not see a meaningful difference between my setup and the template, except that the template works. I am kinda new to Javascript and this whole ecosystem and it's confusing me a bit. What information should I provide so you can more effectively diagnose my issue?

**Comment by DutchEllie** [comment]
I see, of course only after commenting and asking for help once more have I been able to fix the issue, my apologies. So what happened for some reason was that the `vite` package version was not updated in the `package.json` for some reason. I finally noticed the version difference and noticed how `package.json` files under `node_modules` dirs also export stuff. That brought me to the `@sveltejs` package, which was throwing the errors and it said that the vite package was experimental, exporting it under something like `experimental/vite`, rather than `vite`, which was what `vite.config.js` was expecting. I had to manually install this package again by executing `npm install --save-dev @sveltejs/kit` for it to update, something I expected `npm install` to be able to do, which was my mistake. 

Now then, my issue is solved. Maybe this can help someone in the future too who knows. Seems like a weird error and a weird mistake on npm's side? Refusing to update a package? Must have been my mistake somehow anyway
